src/CMakeLists.txt

```diff
@@ -271,20 +240,133 @@ set (NN_SOURCES
 )
 
 if (WIN32)
-    LIST (APPEND NN_SOURCES
```

Comment:
I'm still pretty opposed to this approach.  Part of the reason is that it will be easy for folks to think they're modifying the file list for all platforms, and only be modifying them for one of them.  I agree we can refactor to better than what we had before this change,but I don't think this part of the change set represents an improvement.